packages/core-data/src/resolvers.js

```diff
@@ -382,3 +386,38 @@ export function* getAutosaves( postType, postId ) {
 export function* getAutosave( postType, postId ) {
 	yield controls.resolveSelect( 'core', 'getAutosaves', postType, postId );
 }
+
+/**
+ * Retrieve the frontend template used for a given link.
+ *
+ * @param {string} link  Link.
+ */
+export function* __experimentalGetTemplateForLink( link ) {
```

Comment:
That's a frontend link (whatever that link maybe, it can be a post link, a home page link, an archive page link...) you give  it a link, it gives you the template used for that link.